libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/SFSDKUserSelectionTableViewController.m

```diff
@@ -29,157 +29,279 @@
+    [self.addButton addTarget:self action:@selector(createUser) forControlEvents:UIControlEventTouchDown];
     
-    UITextView *infoLabel = [[UITextView alloc] initWithFrame:topStackViewInner.frame];
-    infoLabel.editable = FALSE;
+    [self.addButton.leftAnchor  constraintEqualToAnchor:self.leftAnchor constant:kHorizontalSpace *3].active = YES;
+    [self.addButton.topAnchor  constraintEqualToAnchor:self.topAnchor constant:kVerticalSpace].active = YES;
+    [self.descriptionLabel.leftAnchor   constraintEqualToAnchor:self.addButton.rightAnchor constant:kHorizontalSpace].active = YES;
+    [self.descriptionLabel.topAnchor  constraintEqualToAnchor:self.topAnchor constant:kVerticalSpace].active = YES;
+}
+
+- (void)createUser {
+    [self.footerDelegate createUser];
+}
+@end
+
+@implementation SFSDKUserSelectionTableViewController
+@synthesize spAppOptions;
+@synthesize listViewDelegate;
+
+- (void)loadView {
+    [super loadView];
+   
+    self.tableView = [self createTableView];
+    self.view.backgroundColor = [UIColor backgroundcolor];
+    self.title = [SFSDKResourceUtils localizedString:@"idpSelectUserTitleLabel"];
+    UIView *headerView = [self createHeaderView];
+    UIView *footerView = [self createFooterView];
+    self.tableView = [self createTableView];
     
-   // infoLabel = 5;
-    UIFont *font = [UIFont boldSystemFontOfSize:18.0];
-    UIFont *normalFont = [UIFont systemFontOfSize:18.0];
-    NSString *appName = [self.options objectForKey:kSFAppNameParam];
+    self.tableView.tableFooterView = footerView;
+    [self.tableView.tableFooterView.bottomAnchor constraintEqualToAnchor:self.tableView.bottomAnchor].active = YES;
+    [self.tableView.heightAnchor constraintEqualToConstant:(self.view.bounds.size.height * 3)/4].active = YES;
+    [self.tableView.widthAnchor constraintEqualToConstant:self.view.bounds.size.width].active = YES;
     
-    if (!appName) {
-        appName = kSFAppNameDefault;
-    }
+    UIStackView *stack = [[UIStackView alloc] initWithFrame:CGRectZero];
+    stack.backgroundColor =  [UIColor backgroundcolor];
+    stack.distribution = UIStackViewDistributionEqualSpacing;
+    stack.alignment = UIStackViewAlignmentCenter;
+    stack.translatesAutoresizingMaskIntoConstraints = NO;
+    stack.spacing = 0;
     
-    NSMutableAttributedString *attributedText = [[NSMutableAttributedString alloc] initWithString:appName  attributes:@{ NSFontAttributeName : font, NSForegroundColorAttributeName : [UIColor redColor] }];
+    [stack setAxis:UILayoutConstraintAxisVertical];
+    [stack addArrangedSubview:headerView];
+    [stack addArrangedSubview:self.tableView];
+    [self.view addSubview:stack];
     
-    NSMutableAttributedString *attributedNormalText = [[NSMutableAttributedString alloc] initWithString:@" is requesting access to users credentials. Select a user from the list" attributes:@{NSFontAttributeName : normalFont , NSForegroundColorAttributeName : [UIColor grayColor] }];
+    if (@available(iOS 11, *)) {
```

Comment:
Can we make this such that it works for both `iOS 10` and `iOS 11` (basically fall back on the `iOS10` API)?